include/tvm/runtime/vm/vm.h

```diff
@@ -256,7 +256,7 @@ class VirtualMachine : public runtime::ModuleNode {
    *
    * This does not begin execution of the VM.
    */
-  void InvokeGlobal(const VMFunction& func, const std::vector<ObjectRef>& args);
+  void InvokeGlobal(const VMFunction& func, Index func_index, const std::vector<ObjectRef>& args);
```

Comment:
Why do we need to pass the function index here?

Reply:
I see the code has been refactored, I think we should store the function index in the function itself so we don't have to pass extra state around and the VMFunctions are sufficient to construct local state.

Reply:
Because VirtualMachine::func_index_ is not updated in the function VirtualMachine::InvokeGlobal like VirtualMachine::pc_ does。
or VirtualMachine::func_index_  should be deleted, I'm not seeing any place that use VirtualMachine::func_index_ in current code。
If VirtualMachine::func_index_ will be used in future。 then VirtualMachine::func_index_ should be updated in VirtualMachine::InvokeGlobal. otherwise, the logic is not right.

Reply:
in VirtualMachine::InvokeGlobal , function PushFrame is pushing a frame which include curernt's code_ ,pc_, func_idx_  into  VirtualMachine::frames_.  current's code_, pc_ is updated in  VirtualMachine::InvokeGlobal. but current func_idx_ is updated in the entire code . I think func_idx_ should be updated in VirtualMachine::InvokeGlobal the same way as pc_ and code_.  So I pass the func_idx to the function VirtualMachine::InvokeGlobal